bentoml/adapters/json_input.py

```diff
@@ -14,18 +14,140 @@
 
 import os
 import json
+import inspect
+import functools
 import argparse
-from typing import Iterable
+from typing import Iterable, List
 from json import JSONDecodeError
 
+import uuid
 import flask
 
 from bentoml.exceptions import BadInput
-from bentoml.marshal.utils import SimpleResponse, SimpleRequest
+from bentoml.types import (
+    HTTPRequest,
+    HTTPResponse,
+    JsonSerializable,
+    AwsLambdaEvent,
+    InferenceTask,
+)
 from bentoml.adapters.base_input import BaseInputAdapter
+from bentoml.adapters.base_output import BaseOutputAdapter
 from bentoml.adapters.utils import concat_list
 
 
+class JsonInferenceTask(InferenceTask):
+    data: str  # json string
+
+
+class InferenceAPI:
```

Comment:
this will merge with the InferenceAPI class here right? https://github.com/bentoml/BentoML/blob/v0.8.4/bentoml/service.py#L52

Reply:
Yep